.github/workflows/build-test.yml

```diff
@@ -0,0 +1,55 @@
+name: Build
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    env:
+      OFFICIAL_BUILD: 'True'
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: 6.0.x
+
+      - name: Restore packages
+        run: dotnet restore
+
+      - name: Build
+        run: dotnet build --no-restore --configuration Debug
+
+      - name: Test
+        run: dotnet test --configuration Debug /p:CollectCoverage=true /p:Threshold=100 /p:ThresholdType=line /p:ThresholdStat=total /p:CoverletOutputFormat=cobertura /p:CoverletOutput=$GITHUB_WORKSPACE/coverage.xml
+
+      - name: Generate Test Coverage Report
+        uses: danielpalme/ReportGenerator-GitHub-Action@5.1.3
+        with:
+          reports: 'coverage.xml'
+          targetdir: 'coveragereport'
+          reporttypes: 'HtmlInline;Cobertura;Badges' # The output formats and scope (separated by semicolon) Values: Badges, Clover, Cobertura, CsvSummary, Html, Html_Dark, Html_Light, HtmlChart, HtmlInline, HtmlInline_AzurePipelines, HtmlInline_AzurePipelines_Dark, HtmlInline_AzurePipelines_Light, HtmlSummary, JsonSummary, Latex, LatexSummary, lcov, MarkdownSummary, MHtml, PngChart, SonarQube, TeamCitySummary, TextSummary, Xml, XmlSummary
+          verbosity: 'Info'
+          tag: '${{ github.run_number }}_${{ github.run_id }}'
+          toolpath: 'reportgeneratortool'
+        if: always()
+
+      - name: Upload Coverage Report Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: CoverageReport
+          path: coveragereport
+        if: always()
```

Comment:
nit: Do we want to follow the new line at end of file recommended by GitHub?

Reply:
Personally, I hate having an empty line at the end of files; but this does seem to unfortunately be more the norm. Even the front-end formatter Prettier seems to auto-add an empty line to files on format, without any configurable way to stop doing it. So, I may just be the odd person out here. 😢